### PR TITLE
[CMake] Make sure that library evolution is set when compiling overlays.

### PR DESCRIPTION
**Pull request by varungandhi-apple**
If library evolution is not set, the generated swiftinterface will be incorrect, leading to further errors downstream.

In direct CMake calls to `stdlib/public/Platform/CMakeLists.txt`, the variable `SWIFT_STDLIB_STABLE_ABI` is not initialized. ~For now, we duplicate the logic in `stdlib/CMakeLists.txt` to make sure that it is initialized properly. We can potentially de-duplicate it later.~ So we add some initialization in `StandaloneOverlay.cmake` which is consumed by `stdlib/public/Platform/CMakeLists.txt`, mimicking the "duplicate" initialization we already have for `SWIFT_ENABLE_MODULE_INTERFACES`.

Fixes rdar://70156840.

### CODE REVIEW

**Comment by varungandhi-apple** [comment]
cc @porglezomp

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

File: cmake/modules/StandaloneOverlay.cmake [hunk 1]
```diff
@@ -58,9 +58,44 @@ set(SWIFT_NATIVE_CLANG_TOOLS_PATH "${TOOLCHAIN_DIR}/usr/bin" CACHE STRING
 set(SWIFT_NATIVE_SWIFT_TOOLS_PATH "${TOOLCHAIN_DIR}/usr/bin" CACHE STRING
   "Path to Swift tools that are executable on the build machine.")
 
+# HACK: This initialization code is duplicated in stdlib/CMakeLists.txt
+
+if("${SWIFT_HOST_VARIANT_SDK}" MATCHES "(OSX|IOS*|TVOS*|WATCHOS*)")
+  # All Darwin platforms have ABI stability.
+  set(SWIFT_STDLIB_STABLE_ABI_default TRUE)
+elseif("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "LINUX")
```
Comment (lorentey) [inline cmake/modules/StandaloneOverlay.cmake:66]: I'm pretty sure that this file is only used to build the overlays for official Apple platform releases, so we won't need this platform switch here -- this only ever applies to Darwin.

Comment (varungandhi-apple) [reply cmake/modules/StandaloneOverlay.cmake]: Removed.

**Comment by varungandhi-apple** [comment]
@swift-ci please smoke test

**Review by lorentey — APPROVED** [review]
👍

**Review by edymtt — APPROVED** [review]

**Pull request merged**